pkg/dns/aws/dns.go

```diff
@@ -503,7 +516,40 @@ func (m *Provider) change(record *iov1.DNSRecord, zone configv1.DNSZone, action
 	// Find the target hosted zone of the load balancer attached to the service.
 	targetHostedZoneID, err := m.getLBHostedZone(target)
 	if err != nil {
-		return fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
+		err = fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
+		if v, ok := record.Annotations[targetHostedZoneIdAnnotationKey]; !ok {
+			return err
+		} else {
+			log.Error(err, "falling back to the "+targetHostedZoneIdAnnotationKey+" annotation", "value", v)
+			targetHostedZoneID = v
+		}
+	}
+	// If this is an upsert, store the target hosted zone id in an
+	// annotation on the DNSRecord CR in case we later on we need the id and
+	// for whatever reason cannot look it up using the AWS API.
+	if action == upsertAction {
+		var current iov1.DNSRecord
+		name := types.NamespacedName{
+			Namespace: record.Namespace,
+			Name:      record.Name,
+		}
+		if err := m.config.Client.Get(context.TODO(), name, &current); err != nil {
+			// Log the error and continue.  The annotation is only
+			// needed as a fallback mechanism, and anyway we might
+			// succeed in adding it on the next upsert.
+			log.Error(err, "failed to get dnsrecord", "dnsrecord", name)
+		} else if _, ok := record.Annotations[targetHostedZoneIdAnnotationKey]; !ok {
```

Comment:
```suggestion
		} else if _, ok := current.Annotations[targetHostedZoneIdAnnotationKey]; !ok {
```

Reply:
Fixed in https://github.com/openshift/cluster-ingress-operator/compare/c1c4ef672c6a88f64ad61f33ffaa30d83b993a44..503e0364c54c9f869b0ea8aaa7175b7bf458dca8.